packages/flutter/test/widgets/list_wheel_scroll_view_test.dart

```diff
@@ -0,0 +1,390 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import '../rendering/mock_canvas.dart';
+import '../rendering/rendering_tester.dart';
+
+void main() {
+  testWidgets('ListWheelScrollView needs positive diameter ratio', (WidgetTester tester) async {
+    try {
+      await tester.pumpWidget(
+        new Directionality(
+          textDirection: TextDirection.ltr,
+          child: new ListWheelScrollView(
+            diameterRatio: -2.0,
+            itemExtent: 20.0,
+            children: <Widget>[],
+          ),
+        )
+      );
+      fail('Expected failure with negative diameterRatio');
+    } on AssertionError {
+      // Exception expected.
```

Comment:
Use `expect(() { ... }, throwsAssertionError);`

Reply:
Also if you want to test the assertion comes from the constructor, no need to call pumpWidget. Just call the constructor. Otherwise you may be catching other asserts and not realising it.

Also, you should probably verify the text of the assert, for the same reason (to make sure you're catching the right one).

Reply:
Left the try catch to assert on the message.